corehq/apps/reports/standard/project_health.py

```diff
@@ -48,49 +49,81 @@ class MonthlyPerformanceSummary(jsonobject.JsonObject):
     active = jsonobject.IntegerProperty()
     performing = jsonobject.IntegerProperty()
 
-    def __init__(self, domain, month, performance_threshold, previous_summary=None):
+    def __init__(self, domain, month, users_filtered_by_group, performance_threshold, previous_summary=None):
         self._previous_summary = previous_summary
+        self._users_filtered_by_group = users_filtered_by_group
```

Comment:
Similar to what Cory suggested - you could only fork the logic here in the constructor, then the changes applied to `_base_queryset` will be available everywhere necessary.  That is:
```python
if users_filtered_by_group:
    # add additional filter to _base_queryset
```
Then you should be able to just always refer to `_base_queryset` without needing to check `users_filtered_by_group`